test/DecentHats.test.ts

```diff
@@ -0,0 +1,826 @@
+import { SignerWithAddress } from '@nomicfoundation/hardhat-ethers/signers';
+import { expect } from 'chai';
+/* eslint-disable-next-line import/no-extraneous-dependencies */
+import { ethers } from 'ethers';
```

Comment:
So that 'can't find ethers namespace' errors was removing type safety. After quite a bit of trial and error of trying things. I found this is the simplest solution. `hardhat` or one of the `hardhat` libraries is adding `ethers` to our `node_modules` but it is not in our `package.json` but adding it there conflicts with the version used in the hard hat environment. 

So doing this allows for it to properly find it and doesn't prevent the other errors from showing up. 

I did not add this to other files from my own as I didn't want to fill up this PR but I can.

Reply:
I dig it, thanks. I'll open up a standalone PR.

Reply:
Well actually there are only two other test files that have this issue. Would you mind just applying this fix to those two other files in your current PR? thanks.